### PR TITLE
Renaming io.realm.internal.Collection to io.realm.internal.OsResults

### PR DESCRIPTION
**Pull request by kneth**
Part III of #4215 

Depends on #4933

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
  * size and order of elements will never be changed after creation.
  * <p>
  * {@link OrderedRealmCollectionSnapshot} is useful when making changes which may impact the size or order of the
- * collection in simple loops. For example:
+ * osResults in simple loops. For example:
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java [hunk 2]
```diff
@@ -199,12 +199,12 @@ public boolean deleteFirstFromRealm() {
...
         realm.checkIfValidAndInTransaction();
-        UncheckedRow row = collection.lastUncheckedRow();
-        return row != null && row.isAttached() && collection.deleteLast();
+        UncheckedRow row = osResults.lastUncheckedRow();
+        return row != null && row.isAttached() && osResults.deleteLast();
     }
 
     /**
-     * This deletes all objects in the collection from the underlying Realm. All objects in the collection snapshot
+     * This deletes all objects in the osResults from the underlying Realm. All objects in the osResults snapshot
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 3]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
      * element {@code e} returned by the iterator, whether it is contained in
-     * the specified {@code Collection}. If this test is positive, then the {@code
+     * the specified {@code OsResults}. If this test is positive, then the {@code
      * remove} method is called on the iterator.
      *
-     * @param collection the collection of objects to remove.
-     * @return {@code true} if this {@code Collection} is modified, {@code false} otherwise.
-     * @throws ClassCastException if one or more elements of {@code collection} isn't of the correct type.
-     * @throws NullPointerException if {@code collection} is {@code null}.
+     * @param collection the osResults of objects to remove.
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 4]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
     }
 
     /**
-     * Removes all occurrences in this {@code Collection} of each object in the
-     * specified {@code Collection}. After this method returns none of the
-     * elements in the passed {@code Collection} can be found in this {@code Collection}
+     * Removes all occurrences in this {@code OsResults} of each object in the
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 5]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
+     * specified {@code OsResults}. After this method returns none of the
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 6]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
+     * elements in the passed {@code OsResults} can be found in this {@code OsResults}
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 7]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
      * anymore.
      * <p>
-     * This implementation iterates over the {@code Collection} and tests each
+     * This implementation iterates over the {@code OsResults} and tests each
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 8]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
      * element {@code e} returned by the iterator, whether it is contained in
-     * the specified {@code Collection}. If this test is positive, then the {@code
+     * the specified {@code OsResults}. If this test is positive, then the {@code
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 9]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
      * remove} method is called on the iterator.
      *
-     * @param collection the collection of objects to remove.
-     * @return {@code true} if this {@code Collection} is modified, {@code false} otherwise.
-     * @throws ClassCastException if one or more elements of {@code collection} isn't of the correct type.
-     * @throws NullPointerException if {@code collection} is {@code null}.
+     * @param collection the osResults of objects to remove.
+     * @return {@code true} if this {@code OsResults} is modified, {@code false} otherwise.
+     * @throws ClassCastException if one or more elements of {@code osResults} isn't of the correct type.
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 10]
```diff
@@ -380,20 +381,20 @@ public boolean remove(Object object) {
...
+     * @throws NullPointerException if {@code osResults} is {@code null}.
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java [hunk 11]
```diff
@@ -151,7 +151,7 @@ public abstract ColumnInfo validateTable(Class<? extends RealmModel> clazz,
      * After being inserted any changes to the original objects will not be persisted.
      *
      * @param realm reference to the {@link Realm} where the objects will be inserted.
-     * @param objects Collection of {@link RealmObject} to insert or update. This must not be empty.
+     * @param objects OsResults of {@link RealmObject} to insert or update. This must not be empty.
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java]: wrong rename

File: realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java [hunk 12]
```diff
@@ -161,7 +161,7 @@ public abstract ColumnInfo validateTable(Class<? extends RealmModel> clazz,
...
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java]: wrong rename

**Comment by kneth** [comment]
@Zhuinden Damn, I thought I checked them all.

**Comment by Zhuinden** [comment]
@kneth hopefully I didn't do any false positives 😄

**Comment by kneth** [comment]
@Zhuinden Looks like you found the right ones - thanks.

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java [hunk 13]
```diff
@@ -204,7 +204,7 @@ public boolean deleteLastFromRealm() {
     }
 
     /**
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java]: I think you should just remove the second `osResults` here.

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java [hunk 14]
```diff
@@ -204,7 +204,7 @@ public boolean deleteLastFromRealm() {
...
-     * This deletes all objects in the osResults from the underlying Realm. All objects in the osResults snapshot
+     * This deletes all objects in the collection from the underlying Realm. All objects in the osResults snapshot
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java]: I think you should just remove the second osResults here.

**Comment by Zhuinden** [comment]
I still can't shake the feeling that if these renames happen, they will be the last PRs before the 4.0 release.

**Comment by beeender** [comment]
Replaced by #5452

**Pull request closed without merging**